### ISSUE TITLE
Notifications menu and contact-search menu broken with certain predefined backgrounds

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/server#42962]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps

1. Go to go to index.php/settings/user/theming
2. Select one of the following default offered backgrounds: All that make the font color go black
  - Pink bubbles
  - Yellow bricks
  - Lion rock hill
  - "the ducks"
  - soft floral
  - morning fog
4. Open the notification menu => Inverted color ❌ 
5. Open the contact search => menu opens in place of the button instead of a popover menu 💥 
![Bildschirmfoto vom 2024-01-19 14-43-44](https://github.com/nextcloud/server/assets/213943/2f8b5efb-8692-42cc-84ea-42ee9948c4c0)

How it looks with any other background:

![Bildschirmfoto vom 2024-01-19 14-43-35](https://github.com/nextcloud/server/assets/213943/cb7c357e-b561-40a0-a362-67d1fcdb01d0)

**Comment by ShGKme** [comment]
- Regression of https://github.com/nextcloud/server/pull/42712

While filter inverts the icon, it also breaks the whole header menu.

Without filter | With filter
---|---
![image](https://github.com/nextcloud/server/assets/25978914/73c56528-ec0e-4bda-86b2-1b315b779821) | ![image](https://github.com/nextcloud/server/assets/25978914/3688823b-e9f7-4411-8628-f8d7fd9f8bf5)

**Comment by szaimen** [comment]
Yeah... I wonder how a simple filter can do this???


https://github.com/nextcloud/server/assets/42591237/e681f702-626f-41f3-a7ec-fae5249a32e7

**Comment by szaimen** [comment]
Yeah, I wonder how a simple filter can do this??? Browser bug maybe?

**Comment by ShGKme** [comment]
> Yeah, I wonder how a simple filter can do this??? Browser bug maybe?

Anyway, we should not invert a whole menu. Only the icon.

**Comment by szaimen** [comment]
> Anyway, we should not invert a whole menu. Only the icon.

I agree.

**Comment by szaimen** [comment]
Fix is in https://github.com/nextcloud/server/issues/42962

I am not really satisfied with the solution but it is what it is...

**Comment by blizzz** [comment]
> Fix is in #42962


Or maybe https://github.com/nextcloud/server/pull/42963

**Comment by szaimen** [comment]
> > Fix is in #42962
> 
> Or maybe #42963

Yes, sorry

**Comment by ChristophWurst** [comment]
It's not just a visual bug. I see a network request to load the content of the menu even before it opens. That should only happen on click.

**Comment by ChristophWurst** [comment]
Test master. The menu is still as broken as it was before for me:

![Bildschirmfoto vom 2024-01-22 12-07-07](https://github.com/nextcloud/server/assets/1374172/d503cb20-5cc3-4625-
bf05-790356d76af9) 

The `POST /contactsmenu/contacts` also happens after page load, before I event click on the contacts menu toggle.

**Comment by szaimen** [comment]
@ChristophWurst apparently someone forgot to recompile css and our CI didnt catch that... See https://github.com/nextcloud/server/pull/43017

**Comment by ChristophWurst** [comment]
> The `POST /contactsmenu/contacts` also happens after page load, before I event click on the contacts menu toggle.

Is still there, even after recompilation.

**Comment by szaimen** [comment]
> > The `POST /contactsmenu/contacts` also happens after page load, before I event click on the contacts menu toggle.
> 
> Is still there, even after recompilation.

See https://github.com/nextcloud/server/issues/43018
